File: manual/code/test_suites/python/python/test_bokeh_circle.py
import unittest
from function_bokeh_circle import Plotter

class TestBokehCirclePlot(unittest.TestCase):
    
    def test_circle_plot_creation(self):
        plotter = Plotter()
        result = plotter.create_circle_plot("[1, 2, 3]", "[4, 5, 6]")
        self.assertEqual(result, "Plot displayed.")
    
    def test_empty_input(self):
        plotter = Plotter()
        result = plotter.create_circle_plot("[]", "[]")
        self.assertEqual(result, "Error: Invalid input format.")
    
    def test_invalid_length(self):
        plotter = Plotter()
        result = plotter.create_circle_plot("[1, 2]", "[3]")
        self.assertEqual(result, "Error: x_values and y_values must have the same length.")
    
    def test_custom_title(self):
        plotter = Plotter()
        result = plotter.create_circle_plot("[1, 2, 3]", "[4, 5, 6]", title="Custom Circle Plot Title")
        self.assertEqual(result, "Plot displayed.")
    
    def test_custom_radius(self):
        plotter = Plotter()
        result = plotter.create_circle_plot("[1, 2, 3]", "[4, 5, 6]", radius=15)
        self.assertEqual(result, "Plot displayed.")
        
if __name__ == '__main__':
    unittest.main()

File: manual/data/java/tests/test_bokeh_circle.java
package com.example;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class test_bokeh_circle {
    @Test
    public void test_circle_plot_creation() {
        function_bokeh_circle plotter = new function_bokeh_circle();
        String result = plotter.create_circle_plot("[1, 2, 3]", "[4, 5, 6]", 10, "Circle Plot");
        assertEquals("Plot displayed.", result);
    }

    @Test
    public void test_empty_input() {
        function_bokeh_circle plotter = new function_bokeh_circle();
        String result = plotter.create_circle_plot("[]", "[]", 10, "Circle Plot");
        assertEquals("Error: Invalid input format.", result);
    }

    @Test
    public void test_invalid_length() {
        function_bokeh_circle plotter = new function_bokeh_circle();
        String result = plotter.create_circle_plot("[1, 2]", "[3]", 10, "Circle Plot");
        assertEquals("Error: x_values and y_values must have the same length.", result);
    }

    @Test
    public void test_custom_title() {
        function_bokeh_circle plotter = new function_bokeh_circle();
        String result = plotter.create_circle_plot("[1, 2, 3]", "[4, 5, 6]", 10, "Custom Circle Plot Title");
        assertEquals("Plot displayed.", result);
    }

    @Test
    public void test_custom_radius() {
        function_bokeh_circle plotter = new function_bokeh_circle();
        String result = plotter.create_circle_plot("[1, 2, 3]", "[4, 5, 6]", 15, "Circle Plot");
        assertEquals("Plot displayed.", result);
    }
}